File: FrontEnd/stock-analyzer/src/app/types/Company.ts
import { BhavInfo } from "./BhavInfo"
import { BulkDeal } from "./BulkDeal"

export type Company = {
    symbol: string,
    fullName: string,
    bhavCopyInfos: BhavInfo[],
    bulkDeals: BulkDeal[],
}

File: Stock Analyzer/Stock Analyzer Domain/Models/Company.cs
using Stock_Analyzer_Domain.Models.Filter;

namespace Stock_Analyzer_Domain.Models
{
  public class Company
  {
    public Guid Id { get; set; }

    public string Symbol { get; set; }

    public ICollection<BhavCopyInfo> BhavCopyInfos { get; set; } = new List<BhavCopyInfo>();

    public ICollection<BulkDeal> BulkDeals { get; set; } = new List<BulkDeal>();

    public List<FilterResult> FilterResults { get; set; } = new List<FilterResult>();

    public Company()
    { }

    public Company(string name)
    {
      this.Symbol = name;
    }
  }
}
